programs/gzip.c

```diff
@@ -425,7 +436,10 @@ decompress_file(struct libdeflate_decompressor *decompressor, const tchar *path,
 	if (ret != 0)
 		goto out_close_out;
 
-	ret = do_decompress(decompressor, &in, &out);
+	if (options->test) /// decompress file, if it succeeds, test went ok, if fails, library generates message anyways
```

Comment:
It would be more self-explanatory to just pass the `options` down to `do_decompress()`.

`ret = do_decompress(decompressor, &in, &out, options);`

Reply:
Understood.

Reply:
That would mean adding one argument to this function.